db/schema_applier.cc

```diff
@@ -885,8 +880,26 @@ future<> schema_applier::update() {
     }
 }
 
-void schema_applier::commit() {
-    // TODO: add copy on write to schema changes
+void schema_applier::commit_on_shard(replica::database& db) {
+    // commit user functions and aggregates
+    _functions_batch.local().commit();
+}
+
+// TODO: move per shard logic directly to raft so that all subsystems can be updated together
+// (requires switching all affected subsystems to 'applier' interface first)
+future<> schema_applier::commit() {
+    auto& sharded_db = _proxy.local().get_db();
+    // Run func first on shard 0
+    // to allow "seeding" of the effective_replication_map
+    // with a new e_r_m instance.
+    co_await sharded_db.invoke_on(0, [this] (replica::database& db) { return commit_on_shard(db); });
```

Comment:
Can't we guaranty that the `commit` is called on shard0?

Reply:
yes we do, this invoke_on is effectively no-op, code originates from different place, I can simplify it, but perhaps in separate commit

Reply:
> yes we do, this invoke_on is effectively no-op, code originates from different place, I can simplify it, but perhaps in separate commit

OK. You can also make next line to be `invoke_on_others` and then you can drop shard0 checking.

Reply:
added already in https://github.com/scylladb/scylladb/pull/20853/commits/102b8ff182cad121e273cbd9b1b9592c08133766